sync2/handler2/handler.go

```diff
@@ -31,12 +32,14 @@ var logger = zerolog.New(os.Stdout).With().Timestamp().Logger().Output(zerolog.C
 // processing v2 data (as a sync2.V2DataReceiver) and publishing updates (pubsub.Payload to V2Listeners);
 // and receiving and processing EnsurePolling events.
 type Handler struct {
-	pMap      sync2.IPollerMap
-	v2Store   *sync2.Storage
-	Store     *state.Storage
-	v2Pub     pubsub.Notifier
-	v3Sub     *pubsub.V3Sub
-	unreadMap map[string]struct {
+	pMap    sync2.IPollerMap
+	v2Store *sync2.Storage
+	Store   *state.Storage
+	v2Pub   pubsub.Notifier
+	v3Sub   *pubsub.V3Sub
+	// user_id+room_id => fnv_hash(last_event_bytes)
+	accountDataMap *sync.Map
```

Comment:
What is the difference between a plain map and a sync.Map? The latter has a mutex built in, and only offers concurrency-aware operations?

Reply:
Basically yes. It is particularly good for a few use cases:
- (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or 
- (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys.

In our case, we will (it's currently on the executor) do 2).